go/tasks/pluginmachinery/k8s/plugin.go

```diff
@@ -35,6 +35,14 @@ type PluginEntry struct {
 	// support the same task type. This must be a subset of RegisteredTaskTypes and at most one default per task type
 	// is supported.
 	DefaultForTaskTypes []pluginsCore.TaskType
+	// Returns a new KubeClient to be used instead of the internal controller-runtime client.
+	NewKubeClient func(ctx context.Context) (pluginsCore.KubeClient, error)
+	// Boolean that indicates if kubernetes resources that this plugin is responsible for should include OwnerReferences.
+	// Ingoring is only useful if resources will be created in a remote cluster.
+	IgnoreOwnerReferences bool
+	// Boolean flag that indicates if a finalizer should be injected.
+	// This will override the `inject-finalizer` set under k8s config.
+	InjectFinalizers *bool
```

Comment:
```suggestion
	overrideInjectOwnerReference *bool
	// Boolean flag that indicates if a finalizer should be injected.
	// This will override the `inject-finalizer` set under k8s config.
	overrideInjectFinalizers *bool
```

Reply:
Actually these will need to be public.

Reply:
Sorry yeah I didn't mean to make them private... more about the naming and meaning...

Reply:
I understood you 😄 and yea naming is always hard 😅 

Updated the PR but smth is off with copilot docker image pushing the image... seems like perm stuff